### DISCUSSION TITLE
Searching for words without `.` doesn't match any string in the index.

### DISCUSSION THREAD

**Discussion by litcodeur** [meilisearch/product#151] [Feedback & Feature Proposal]
**Describe the bug**
Searching for words without `.` doesn't match any string in the index. For ex: searching for `nodejs` in the index containing the word `node.js` doesn't match. 

**Easiest way to reproduce**
1. Go to https://docs.meilisearch.com/
2. Search for `nodejs` (shows in no result found)
3. Search for `node.js` (shows a match)

**Additional context**
I am trying to create a blog with meilisearch and this is the only thing that kinda bothers me. I think that most people trying to search for text like `node.js` or `socket.io` won't add the '.' while typing by still expect the results to match. Is there a way to configure the server to ignore the `.` and match the rest of the characters anyways?

**Comment by CaroFG** [comment]
Hi! It is expected behavior as `.` are considered hard spaces (check out [this](https://docs.meilisearch.com/reference/under_the_hood/datatypes.html#string) doc's section) so `node.js` is considered two words instead of one. You'll get results by typing: `node.js` and `node js` but none for `nodejs`

AFAIK there is no way of changing that behavior, but I'll let my colleagues confirm that

**Comment by litcodeur** [comment]
thanks!

**Comment by curquiza** [comment]
Hello @shah62, your feedback is a nice improvement we would like to add to MeiliSearch. The whole core team is currently working on a new search engine, which means we don't have the time for this kind of improvement for the next release (v0.21.0). It will indeed involve consideration about how to implement this and improve our tokenizer.
But we really like to add this dot (`.`) handling in the future, so I re-open the issue if you don't mind 🙂

**Comment by gmourier** [comment]
Thank you @shah62! We have moved the feature proposal requests to the product repository in order to centralize all the preparatory discussions there to better understand your needs.

Indeed, the `.` is considered as a hard separator and is therefore not taken into account when searching for a term that does not contain it as in your example.

We could give the possibility to the user to customize the hard separators from the settings or also propose a set of specific tokenizers so that it is possible to customize the internal behavior of the engine for very specific needs. 

@ManyTheFish I'd love to hear your thoughts on this! What do you think?

**Comment by gmourier** [comment]
Related to #160

**Reply by curquiza** [reply]
@gmourier @shah62 
I'm discussing with @ManyTheFish, we think it's not related to #160 and already fixed in [v1.2](https://github.com/meilisearch/meilisearch/releases/v1.2.0)

**Reply by macraig** [reply]
@curquiza was it fixed in v1.2? If so can the discussion be closed?

**Reply by curquiza** [reply]
Just tested, it works 😊
